src/libstd/sys/wasi/stdio.rs

```diff
@@ -13,8 +13,12 @@ impl Stdin {
     }
 
     pub fn read(&self, data: &mut [u8]) -> io::Result<usize> {
+        self.read_vectored(&mut [IoVecMut::new(data)])
```

Comment:
are vectored reads somehow more "fundamental" on wasi? might be worth a comment?

Reply:
They are yeah where at the syscall layer there's only vectored reads/writes, but this is in general just deduplicating code